### PR TITLE
Add BaseSpan interface to make Span and BlankSpan compatible

### PR DESCRIPTION
**Pull request by mayurkale22**
issue #312

### CODE REVIEW

File: tests/unit/trace/test_base_span.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        span = BaseSpan()
+
+        with self.assertRaises(NotImplementedError):
+            span.add_link(None)
+
+    def test_iter_abstract(self):
+        span = BaseSpan()
+
+        with self.assertRaises(NotImplementedError):
+            list(iter(span))
```
Comment (tgermain) [inline tests/unit/trace/test_base_span.py]: a test for `__enter__` and `__exit__` might be missing. Something like that for instance :
```python
from unittest import mock

@mock.patch(BaseSpan, "__enter__")
@mock.patch(BaseSpan, "__exit__")
def test_context_manager(self, m_exit, m_enter)
    span = BaseSpan()
    with span:
        pass
    self.assertTrue(m_enter.called)
    self.assertTrue(m_exit.called)
```

File: opencensus/trace/blank_span.py [hunk 2]
```diff
@@ -15,9 +15,10 @@
 from opencensus.trace import time_event as time_event_module
 from opencensus.trace.span_context import generate_span_id
 from opencensus.trace.tracers import base
+from opencensus.trace.base_span import BaseSpan
```
Comment (liyanhui1228) [inline opencensus/trace/blank_span.py]: Sort the imports, and import module instead of class.

File: tests/unit/trace/test_base_span.py (hunk 1 above)
Comment (mayurkale22) [reply tests/unit/trace/test_base_span.py]: Done

File: opencensus/trace/blank_span.py (hunk 2 above)
Comment (mayurkale22) [reply opencensus/trace/blank_span.py]: Done

**Comment by mayurkale22** [comment]
Fixes #312

**Comment by davewalkexpel** [comment]
@liyanhui1228 Hi, can a release be cut that included this? It would really help me. Thanks!

**Comment by mayurkale22** [comment]
@davewalkexpel Unfortunately no. But we might do a patch release soon.

**Pull request merged**